include/lbann/layers/layer.hpp

```diff
@@ -203,6 +202,11 @@ class Layer {
   /** Get child layers. (const) */
   inline const std::vector<const Layer*>& get_child_layers() const { return m_child_layers; }
 
+  inline int find_layer_index(const Layer* l) const {
+    return (std::find(m_child_layers.begin(),
+                      m_child_layers.end(),
+                      l) - m_child_layers.begin()); }
```

Comment:
This should return `size_t` -- can't have negative indices! Also, I would recommend using `std::distance` here (can't do multiline suggestions... thanks GH...):
```cpp
inline size_t find_layer_index(const Layer* l) const {
    return std::distance(m_child_layers.begin(),
        std::find(m_child_layers.begin(),
                      m_child_layers.end(),
                      l)); }
```
This suggestion is more of a "straight refactor"; in actuality, you may want to assert that you have child layers (otherwise you'll return `0`, which _is_ a valid index but not what you mean) and that the search is successful, or you'll get `size()`, which is _not_ a valid index. Perhaps code like:
```cpp
inline size_t find_layer_index(const Layer* l) const {
    LBANN_ASSERT(m_child_layers.size());
    auto it = std::find(m_child_layers.begin(), m_child_layers.end(), l);
    LBANN_ASSERT(it != m_child_layers.size());
    return std::distance(m_child_layers.begin(), it);
}
```
(Note: `LBANN_ASSERT` is not a real thing, but it should be.)